src/profilers/time-profiler.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as delay from 'delay';
+
+import {perftools} from '../profile';
+import {serializeTimeProfile} from './profile-serializer';
+
+const timeProfiler = require('bindings')('time_profiler');
+
+export class TimeProfiler {
+  // True when the profiler is actively profiling.
+  private profiling: boolean;
+
+  // samplingInterval is average time in microseconds between samples.
+  constructor(private samplingInterval: number) {
```

Comment:
samplingInterval -> intervalMicros. Please always include the abbreviation of the time unit when the type is a generic type like "number" (rather than something specific like "Duration"). E.g. fooMicros, fooNanos and so on.